TundraTD/Assets/Scripts/City/Building/ElementalTowerUI.cs

```diff
@@ -0,0 +1,97 @@
+using City.Building.Upgrades;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace City.Building
+{
+    public class ElementalTowerUI : MonoBehaviour
+    {
+        [SerializeField] private TowerUpgradeLevel[] upgradeLevels;
+        [SerializeField] private Image[] upgradeLevelIndicators;
+        [SerializeField] private Text allUpgradesBoughtPage;
+
+        private ElementalTower _elementalTower;
+        private Sprite _achievedLevelIndicator;
+        private Sprite _unachievedLevelIndicator;
+        
+        private void Start()
+        {
+            gameObject.SetActive(false);
+            allUpgradesBoughtPage.gameObject.SetActive(false);
+            _achievedLevelIndicator = Resources.Load<Sprite>("UpgradeIcons/Green");
+            _unachievedLevelIndicator = Resources.Load<Sprite>("UpgradeIcons/White");
+
+            foreach (var levelIndicator in upgradeLevelIndicators)
+                levelIndicator.sprite = _unachievedLevelIndicator;
+                
+            foreach (var upgradeLevel in upgradeLevels)
+                upgradeLevel.gameObject.SetActive(false);
+        }
+
+        private void OnEnable()
+        {
+            if (_elementalTower is null)
+                return;
+            
+            for (int i = 0; i < upgradeLevels.Length; i++)
+            {
+                if (i != _elementalTower.TowerUpgradeLevel - 1)
+                    upgradeLevels[i].gameObject.SetActive(false);
+                else
+                    upgradeLevels[i].gameObject.SetActive(true);
```

Comment:
Well, this construction can be simplified with following expression:
```suggestion
                    upgradeLevels[i].gameObject.SetActive(i == _elementalTower.TowerUpgradeLevel - 1);
```